### PR TITLE
Added Atmel mbed hal

### PR DESCRIPTION
**Pull request by Parthasarathy**

### CODE REVIEW

**Comment by Parthasarathy** [comment]
@0xc0170  Any Update on this pull request ? 
We are getting several request from customers not able to setup target dependency due to this.

File: module.json [hunk 1]
```diff
@@ -44,6 +44,9 @@
     },
     "arm-ssg": {
       "mbed-hal-arm-ssg": "^1.0.0"
+    },
+    "atmel": {
+      "mbed-hal-atmel": "^0.1.0"
```
Comment (0xc0170) [inline module.json]: This should be ~0.1.0

**Comment by 0xc0170** [comment]
@Parthasarathy I'll have a look at the yotta modules tree, will report back

File: module.json (hunk 1 above)
Comment (Parthasarathy) [inline module.json]: @0xc0170  
I hope  ^ will allow me to upgrade version 0.X.X.
On other hand ~ will allow only 0.1.X right ?

Comment (Parthasarathy) [inline module.json]: @0xc0170  Referred http://yottadocs.mbed.com/reference/module.html
I am seeing different behaviour . The following dependencies downloads mbed-hal-atmel 0.1.0 but latest version available is "0.1.1". If I change it as ">0.1.0" it gets 0.1.1.  Any idea ?

atmel": {
      "mbed-hal-atmel": "^0.1.0"
}

Comment (0xc0170) [inline module.json]: From the link you referenced 
> ~1.2.3: any version with the same major and minor versions, and an equal or greater patch version.

this is what you want to achieve here for module with 0.x.x . Why mbed-hal-atmel is 0.x.x , not 1.x.x?

Comment (Parthasarathy) [inline module.json]: Thanks. 0.1.0 is for initial version I can update it to 1.0.0

**Comment by 0xc0170** [comment]
The modules should contain homepage and author, and if possible repository url, look at the module.json file in this repository.

It might be useful to provide yt ls for example blinky for instance, to get the tree for review plus at least one target name, which we could use to fetch that tree which this changeset introduces.

Please use `` ~0.x.x `` (rebase your PR here).

**Comment by Parthasarathy** [comment]
I will create new pull request

**Pull request closed without merging**